### ISSUE TITLE
Request: Texture creation, support for overridable internal_format value

### ISSUE THREAD

**Issue by pavelkrupala** [moderngl/moderngl#424]
**A simple use-case of the new feature**

Currently Texture is created in Texture.cpp@120 by calling 
```
gl.TexImage2D(texture_target, 0, internal_format, width, height, 0, base_format, pixel_type, buffer_view.buf);
```
 and internal_format is being decided from @96: ```int internal_format = data_type->internal_format[components];```

However there are bunch more internal formats to use then the defined, namely:

GL_COMPRESSED_RGB_S3TC_DXT1_EXT, GL_COMPRESSED_RGBA_S3TC_DXT1_EXT, GL_COMPRESSED_RGBA_S3TC_DXT3_EXT, and GL_COMPRESSED_RGBA_S3TC_DXT5_EXT
GL_COMPRESSED_SRGB_ALPHA_S3TC_DXT1_EXT, GL_COMPRESSED_SRGB_ALPHA_S3TC_DXT3_EXT, and GL_COMPRESSED_SRGB_ALPHA_S3TC_DXT5_EXT.
but most importantly these:
GL_SRGB8: sRGB image with no alpha.
GL_SRGB8_ALPHA8: sRGB image with a linear Alpha.

**Describe the solution you'd like**

It would be cool if I could provide my own internal format by setting it in the constructor function by specifying **internal_format** by myself:
```
self.output_map_texture = ctx3d.texture(
    out_spec.getSize(), out_spec.components, dtype=out_spec.glformat, internal_format=GL_SRGB8_ALPHA8, data=image_data.tobytes()
)
```
**Version info**

```
moderngl 5.6.2
--------------
vendor: NVIDIA Corporation
renderer: GeForce GTX 970/PCIe/SSE2
version: 3.3.0 NVIDIA 456.71
python: 3.7.4 (tags/v3.7.4:e09359112e, Jul  8 2019, 20:34:20) [MSC v.1916 64 bit (AMD64)]
platform: win32
code: 330
```

**Comment by tpwrules** [comment]
Relatedly, if a custom base_format can be provided, then it is possible to use normalized formats other than unsigned bytes (e.g. signed bytes or unsigned shorts).

**Comment by einarf** [comment]
@tpwrules Thanks for mentioning normalized integers. This is already supported in the `next` branch and should be easy to port into 5.6.x.

The string format for these are: (These are for the format valditaion in moderngl-window that supports both versions)

```python
{
    # Normalized float
    "nf": BufferFormat("nf", 1, 4),
    "nf1": BufferFormat("nf1", 1, 1),
    "nf2": BufferFormat("nf2", 1, 2),
    "nf4": BufferFormat("nf4", 1, 4),
    # Normalized uint
    "nu": BufferFormat("nu", 1, 4),
    "nu1": BufferFormat("nu1", 1, 1),
    "nu2": BufferFormat("nu2", 1, 2),
    "nu4": BufferFormat("nu4", 1, 4),
    # Normalized int
    "ni": BufferFormat("ni", 1, 4),
    "ni1": BufferFormat("ni1", 1, 1),
    "ni2": BufferFormat("ni2", 1, 2),
    "ni4": BufferFormat("ni4", 1, 4),
}
```

**Comment by einarf** [comment]
Fixed in 6aa547879276e80431af4162b23794a9b2c0a68f
